addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -10,29 +9,47 @@ var wUtils = require('website.utils');
 var _t = core._t;
 
 
-var mass_mailing_common = options.Class.extend({
+options.registry.mailing_list_subscribe = options.Class.extend({
     popup_template_id: "editor_new_mailing_list_subscribe_button",
     popup_title: _t("Add a Newsletter Subscribe Button"),
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Allows to select mailing list.
+     *
+     * @see this.selectClass for parameters
+     */
     select_mailing_list: function (previewMode, value) {
         var self = this;
         var def = wUtils.prompt({
             'id': this.popup_template_id,
             'window_title': this.popup_title,
             'select': _t("Newsletter"),
-            'init': function (field) {
+            'init': function (field, dialog) {
                 return rpc.query({
-                        model: 'mail.mass_mailing.list',
-                        method: 'name_search',
-                        args: ['', []],
-                        context: self.options.recordInfo.context,
-                    });
+                    model: 'mail.mass_mailing.list',
+                    method: 'name_search',
+                    args: ['', [['is_public', '=', true]]],
+                    context: self.options.recordInfo.context,
+                }).then(function (data) {
+                    if (!data.length) {
+                        $(dialog).find('.btn-primary').prop('disabled', true);
```

Comment:
You can write `$(dialog).find('.btn-primary').prop('disabled', !data.length);` instead of the `if`